sidecar/dkg/dkg.go

```diff
@@ -15,21 +15,26 @@ import (
 	"time"
 )
 
-type DKGCoordinator struct {
+type Protocol interface {
+	RunDKG(identities []crypto.Identity, sessionID []byte, keypair crypto.Keypair) (*dkg.Result, error)
+	ProcessPacket(packet api.SidecarDKGPacket) error
+}
+
+type Coordinator struct {
 	publicURL string
 	board     *DKGBoard
 	scheme    crypto.ThresholdScheme
 }
 
-func NewDKGCoordinator(publicURL string, scheme crypto.ThresholdScheme) DKGCoordinator {
-	return DKGCoordinator{
+func NewDKGCoordinator(publicURL string, scheme crypto.ThresholdScheme) Coordinator {
+	return Coordinator{
```

Comment:
Shouldn't this return a pointer given the rest of the functions here are taking point receivers?